plugins/ltac2/tac2core.ml

```diff
@@ -1411,7 +1411,21 @@ let () =
     let env = ist.Genintern.genv in
     let sigma = Evd.from_env env in
     let warn = if !Ltac_plugin.Tacintern.strict_check then fun x -> x else Constrintern.for_grammar in
-    let _, pat = warn (fun () ->Constrintern.intern_constr_pattern env sigma ~as_type:false c) () in
+    let expected_vars = Tac2quote.pattern_vars c in
+    let patvars, pat =
+      warn (fun () -> Constrintern.intern_constr_pattern env sigma ~as_type:false c) ()
+    in
+    (* We rely on the non-interned pattern variables in Tac2quote so we need to error if they're forgotten *)
+    let patvars = Id.Map.of_list (List.map (fun v -> v, ()) patvars) in
+    let _ : Util.Empty.t Id.Map.t = Id.Map.merge (fun id expect real ->
+        match expect, real with
+        | None, _ -> assert false
```

Comment:
The error is because binders (eg `fun x => ...`)  get included in `patvars` but are not in `expected_vars`
I don't think I properly understand how this stuff works

Reply:
Does this have anything to do with how binder names are captured in Ltac1 patterns?

Reply:
They're the same patterns I think